arithmetization/src/main/java/net/consensys/linea/zktracer/module/rlptxrcpt/RlpTxnRcpt.java

```diff
@@ -46,45 +48,36 @@
 import org.hyperledger.besu.evm.log.Log;
 import org.hyperledger.besu.evm.log.LogsBloomFilter;
 
-public class RlpTxnRcpt implements Module {
-  private final TxnData txnData;
+@Accessors(fluent = true)
+@RequiredArgsConstructor
+public class RlpTxnRcpt implements StatefullModule<RlpTxrcptOperation> {
   private static final Bytes BYTES_RLP_INT_SHORT = Bytes.minimalBytes(RLP_PREFIX_INT_SHORT);
   private static final Bytes BYTES_RLP_LIST_SHORT = Bytes.minimalBytes(RLP_PREFIX_LIST_SHORT);
 
-  private int absLogNum = 0;
-  @Getter public StackedList<RlpTxrcptChunk> chunkList = new StackedList<>();
+  @Getter
+  private final StackedList<RlpTxrcptOperation> operations =
+      new StackedList<>(LINEA_MAX_NUMBER_OF_TRANSACTIONS_IN_BATCH, 1);
 
-  public RlpTxnRcpt(TxnData txnData) {
-    this.txnData = txnData;
-  }
+  private int absLogNum = 0;
 
   @Override
   public String moduleKey() {
     return "RLP_TXN_RCPT";
   }
 
-  @Override
-  public void enterTransaction() {
-    this.chunkList.enter();
-  }
-
-  @Override
-  public void popTransaction() {
-    this.chunkList.pop();
-  }
-
   @Override
   public void traceEndTx(TransactionProcessingMetadata txMetaData) {
-    RlpTxrcptChunk chunk =
-        new RlpTxrcptChunk(
+    final RlpTxrcptOperation operation =
+        new RlpTxrcptOperation(
             txMetaData.getBesuTransaction().getType(),
             txMetaData.statusCode(),
             txMetaData.getAccumulatedGasUsedInBlock(),
             txMetaData.getLogs());
-    this.chunkList.add(chunk);
+    operations.add(operation);
   }
 
-  public void traceChunk(final RlpTxrcptChunk chunk, int absTxNum, int absLogNumMax, Trace trace) {
+  public void traceChunk(
```

Comment:
I guess we should rename traceChunk methods with traceOperation, right ?

Reply:
yes